app/gui/src/controller/searcher.rs

```diff
@@ -534,6 +538,8 @@ impl Searcher {
             Mode::NewNode { source_node: Some(node), .. } => ThisNode::new(node, &graph.graph()),
             _ => None,
         });
+        let list_builder_with_favorites =
+            component_list_builder_with_favorites(graph.component_groups(), database.clone_ref());
```

Comment:
1. Please remember about splitting expressions in such a way that they are one liners if possible.
2. Do we need to. clone ref the database? It seems strange to me that this struct has two fields and both fields contain reference to the same thing. Maybe its needed, but still feels like a strange desing to me.